objax/util/util.py

```diff
@@ -28,11 +30,30 @@
 
 class EasyDict(dict):
     """Custom dictionary that allows to access dict values as attributes."""
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.__dict__ = self
 
 
+class Renamer:
+    def __init__(self,
+                 rules: Union[Dict[str, str], Tuple[Pattern[str], str], Callable[[str], str]],
```

Comment:
I think second option for `rules` type should be `List[Tuple[Pattern[str], str]]` or `Sequence[Tuple[Pattern[str], str]]` instead of `Tuple[Pattern[str], str]`

`Tuple[Pattern[str], str]` allows user to perform substitutions which match only one pattern, while I would expect that it's likely that user may have multiple patterns for substitution.

Also, `Dict[str, str]` allows user to pass dictionary with multiple substitutions, `Callable` allows to pass a function which can do arbitrary things. Compared to them `Tuple[Pattern[str], str]` looks like somewhat limiting.

Reply:
I agree `Sequence` is a lot nicer, and will do it in the final implementation if design is satisfactory.